repair/repair_sync_boundary.hh

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2018-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include "dht/i_partitioner.hh"
+#include "position_in_partition.hh"
+
+// Represent a position of a mutation_fragment read from a flat mutation
+// reader. Repair nodes negotiate a small range identified by two
+// repair_sync_boundary to work on in each round.
+struct repair_sync_boundary {
```

Comment:
I understood why you wanted to move things out of the source file (repair.cc) and into header files to make them testable, but I didn't understand why you wanted to move things from repair.hh into separate header files. What does this movement get us?
I'm not strongly opposed to this move - it's nothing wrong with it per se - it's just that I don't understand the gain.

Reply:
Bundling unrelated classes in a cc file can be helpful for compilation times (though it's sometimes complicated to evaluate the gains). Bundling classes in a hh file makes sense only if the classes have to live together (e.g. a container + its iterator) or they're always used together. The latter was the case before my patch, but after splitting the repair_writer to a separate header (for testing), the sync boundary can be either separate or should live with the repair_writer that's actually using it. Should we want to have sync boundary unit tests in the future, it would be better to have a separate header for it instead of having to include repair_writer. In the case of this specific class/file split, it doesn't matter much either way. This becomes problematic when headers organically grow like that and you end up with a header that has 3k lines of code, 50 classes inside and is included by half the codebase (see my readers directory change from a couple weeks back - modifying a super simple reader used in one cc file recompiled a third of scylla).